src/lib/button/button.ts

```diff
@@ -141,13 +141,17 @@ export class MdButton implements OnDestroy {
 
   _setElementColor(color: string, isAdd: boolean) {
     if (color != null && color != '') {
-      this._renderer.setElementClass(this._getHostElement(), `mat-${color}`, isAdd);
+      if (isAdd) {
```

Comment:
This one is used in a couple of other places, as well as inside the `button.html` so it may be more convenient to keep it. Otherwise we'd need to do `this._elementRef.nativeElement` all over the place.